client/src/components/_common/CustomMessage/CustomMessage.jsx

```diff
@@ -0,0 +1,66 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { SectionMessage } from '_common';
+import { useDispatch, useSelector } from 'react-redux';
+import styles from './CustomMessage.module.scss';
+
+/**
+ * A message which, is created by the admin. Like IntroMessage, when dismissed, will not appear again.
+ *
+ * _This message is designed for custom messages from the admin.
+ *
+ * @example
+ * // message with identifier
+ * <CustomMessage
+ *   componentName={identifierForMessageLikeRouteName}
+ * >
+ * </CustomMessage>
+ */
+function CustomMessage({ componentName }) {
+  const dispatch = useDispatch();
+  const messages = useSelector((state) => {
+    return state.customMessages
+      ? state.customMessages.messages.filter((message) => {
+          return message.unread && message.template.component === componentName;
+        })
+      : [];
+  });
+
+  function onDismiss(dismissMessage) {
+    const newCustomMessages = {
+      messages: messages.map((message) => {
+        message.unread =
+          message.template.id === dismissMessage.template.id
+            ? false
+            : message.unread;
+        return message;
+      }),
+    };
+    dispatch({
+      type: 'SAVE_CUSTOM_MESSAGES',
+      payload: newCustomMessages,
+    });
+  }
+
+  return messages.map((message) => {
+    const template = message.template;
+    return (
+      <div key={template.id} className={styles.message}>
+        <SectionMessage
+          type={template.message_type}
+          canDismiss={template.dismissible}
+          onDismiss={() => onDismiss(message)}
```

Comment:
Not a part of this PR, but I noticed the close button is not vertically centered
<img width="105" alt="Screen Shot 2022-03-31 at 11 18 21 PM" src="https://user-images.githubusercontent.com/20326896/161194360-c43eb1f0-dae2-45ed-9584-e15257fdd127.png">

Reply:
That problem will likely go away after [FP-546](https://jira.tacc.utexas.edu/browse/FP-546).

<sub>Via Inspector, I solved this by adding `display: flex;` to the button. And the new Button component has `display flex` on it. So, yay!</sub>